authapi/views.py

```diff
@@ -158,16 +158,52 @@ class TeamPermissionViewSet(
     '''Nested viewset to add and remove permissions from teams.'''
     queryset = SeedPermission.objects.all()
     serializer_class = PermissionSerializer
+    permission_classes = (permissions.TeamPermissionPermission,)
 
-    def create(self, request, parent_lookup_seedteam=None, **kwargs):
+    def check_team_permissions(self, request, teamid, orgid=None):
+        old_method = request.method
+        request.method = 'GET'
```

Comment:
We want to check whether the user has read access to the team that they're trying to add/remove the permission for. So we set the request method to `GET` to emulate that the user is trying to get a team's details, and then pass it on with the team object to `has_object_permission` on the permission class, which will tell us whether the user has permission to view the team or not.

So basically, we're emulating a get request to `/teams/<team_id>`, and seeing if the user has permission. If they do, then we allow them to add/remove the permission (pending the PermissionPermission class, which happens automatically), and if they don't, we deny them permission.

I'm not sure if there is a better/cleaner way of doing this, but I couldn't find any.